pkg/config/config.go

```diff
@@ -719,6 +702,7 @@ func (c *config) Load(path string, fsys fs.FS) error {
 		return err
 	}
 	c.Remotes = make(map[string]baseConfig, len(c.Overrides))
+	duplicatedRemotesProjectsIds := make(map[string][]string)
```

Comment:
I'd suggest moving this check to `GetRemoteOverride` because the logic is closely related.

Reply:
I wonder, it felt weird to split the validation logic in the getter since this is where we check for the other fields. And project_id in my mind is just one more field with custom cross fields (across remotes) validation logic.

I can move it, but I think might be best to see thoses issues when your config is parsed and validated, not when you actually try to access one of the declaration. After the Load/Validate, your config is correct, the access logic shouldn't worry about global config consistency / validation.

Reply:
hmm the way I see it, when calling `GetRemoteByProjectRef`, a reasonable expectation is to throw an error if there are multiple remotes with the same matching id.

What you said about validating the config once at load time also makes sense. But perhaps we need something stronger than a warning? Because the iteration order of golang map is randomized. So it would be very surprising for the user if `GetRemoteByProjectRef` returns different config on separate calls.

Reply:
Maybe we can throw an error if duplicates are seen at validation time in that case, keep the warning for "missing" ones that are just undeclared.

And have the `GetRemoteProjectRef` only worry about throwing an error if the requested remote isn't present (and is the same as the base project id maybe ?).

That way user get early feedback about messing up their config, we avoid inconsistent behavior at runtime (the map iteration) and we keep validation in a single place. WDYT ?